src/main/java/org/apache/sling/commons/threads/impl/ThreadPoolMetricsGauges.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with this
+ * work for additional information regarding copyright ownership. The ASF
+ * licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.sling.commons.threads.impl;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.management.DynamicMBean;
+import javax.management.JMException;
+import javax.management.MBeanAttributeInfo;
+import javax.management.MBeanInfo;
+
+import org.apache.sling.commons.metrics.Gauge;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+class ThreadPoolMetricsGauges {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ThreadPoolMetricsGauges.class);
+
+    @SuppressWarnings("unchecked")
+    private static final Map<Class<?>, Class<?>> wrapperToPrimitive = new HashMap() {
+        {
+            put(Integer.class, int.class);
+            put(Long.class, long.class);
+            put(Boolean.class, boolean.class);
+        }
+    };
+
+    @SuppressWarnings("unchecked")
+    private static final Map<Class<?>, Object> clazzToNvl = new HashMap() {
```

Comment:
I would suggest dropping `nvl`, it's specific to Oracle ( I guess :-) ). Maybe something around default value?

Reply:
good catch, it's not null value, but rather value in case of exception, so renamed it according to the suggestion